ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/popup-preview/popup-preview.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import withListenTo, { WithBackboneProps } from '../backbone-container'
+
+import { Metacard } from '.'
+
+const sanitize = require('sanitize-html')
+const PopupPresentation = require('./presentation').default
+
+const NO_PREVIEW = 'No preview text available.'
+const STATUS_OK = 200
+
+const TOP_OFFSET = 60
+
+const br2nl = (str: string) => {
+  return str.replace(/<br\s*\/?>/gm, '\n')
+}
+
+const sanitizeHeader = (header: string) => {
+  if (typeof header === 'string') {
```

Comment:
❓ Do you need to check the type here if it's enforced in the method signature?

Reply:
good point. I'll take it out